tugas/webapp/src/tasks/v-dom.ts

```diff
@@ -1,8 +1,181 @@
+import { loading } from '../performance/reducer';
+import { add, getList, getWorkersList } from './async-action';
+import { TaskList } from './component/task-list';
+import { clearErrorAction, errorAction, store$ } from './store';
 
 new Vue({
   el: '#app',
+  components: {
+    'task-list': TaskList,
+  },
   render(createElement: CreateElement) {
-    return createElement('h1', 'Ini vue component');
+    const workers: VNode[] = [];
+    for (const worker of this.workers) {
+      workers.push(createElement('option', {
+        domProps: {
+          value: worker.id,
+          text: worker.name,
+        }
+      }));
+    }
+    return createElement('div', [
+      createElement('ul', [
+        createElement('li', [
+          createElement('a', {
+            domProps: {
+              href: 'worker.html',
+              innerText: 'pekerja'
+            }
+          })
+        ]),
+        createElement('li', [
+          createElement('a', {
+            domProps: {
+              href: 'tasks.html',
+              innerText: 'pekerjaan'
+            }
+          })
+        ]),
+        createElement('li', [
+          createElement('a', {
+            domProps: {
+              href: 'performance.html',
+              innerText: 'kinerja'
+            }
+          })
+        ]),
+      ]),
+      createElement('hr'),
+      createElement('p', {
+        class: 'error',
+      }, this.error),
+      createElement('p', {
+        class: 'primary',
+        style: {
+          display: this.showLoading
+        }
+      }, 'memuat...'),
+      createElement('form', {
+          on: {
+            submit: this.addNewTask,
+          }
+        }, [
+        createElement('label', 'Tugas'),
+        createElement('br'),
+        createElement('textarea', {
+          domProps: {
+            cols: 30,
+            rows: 3,
+            placeholder: "deskripsi pekerjaan",
+          },
+          on: {
+            change: (e) => {
+              this.task.job = e.target.value;
+            }
+          }
+        }),
+        createElement('br'),
+        createElement('label', 'Pekerja'),
+        createElement('br'),
+        createElement('select', {
+          on: {
+            change: (event) => {
+              this.task.assignee = event.target.value
+            }
+          }
+        }, workers),
+        createElement('br'),
+        createElement('label', 'Lampiran'),
+        createElement('br'),
+        createElement('input', {
+          domProps: {
+            type: 'file', 
+          },
+          on: {
+            input: (event) => {
+              this.task.attachment = event.target.files;
+            }
+          }
+        }),
+        createElement('br'),
+        createElement('button', 'kirim'),
+        createElement('hr'),
+        createElement('h4', 'Daftar Tugas'),
+        createElement('task-list', { props: { tasks: this.tasks } }),
+      ])
+    ]);
+  },
+  data: {
```

Comment:
lebih recomend dibikin 
```js
data(){
return {
data: {},
...
}
}
```

untuk menghindari warning vue

Reply:
```warning
The "data" option should be a function that returns a per-instance value in component definitions.
```

Reply:
oke mas, saya coba ganti. di tempat saya ga keluar warning soalnya. kenapa ya ?

Reply:
https://stackoverflow.com/questions/48176345/what-is-difference-between-data-and-data-in-vue-js 

kalo utk reusable component baru keluar warningnya. kalo buat di root gamasalah sih hehe

Reply:
oh begitu, saya memang di component ga pake data. Makasih mas, bisa untuk menghindari warning di kemudian hari hehehe 👍